public/js/components/FormFields/SearchFields/SearchSuggestions.js

```diff
@@ -0,0 +1,156 @@
+import React, {PropTypes} from 'react';
+import { searchAtoms } from '../../../services/capi';
+import SearchTextInput from './SearchTextInput';
+
+const noop = () => {};
+
+export default class SearchSuggestions extends React.Component {
+  static propTypes = {
+    placeholder: PropTypes.string,
+    filters: PropTypes.object,
+    onSelect: PropTypes.func
+  };
+
+  initialState = 'idle';
+  
+  machine = {
+    idle: {
+      KEY: 'type'
+    },
+    type: {
+      KEY: 'type',
+      TIMEOUT: 'search',
+      ESC: 'idle'
+    },
+    search: {
+      KEY: 'type',
+      RESOLVE: 'results',
+      REJECT: 'error'
+    },
+    results: {
+      KEY: 'type',
+      ESC: 'idle',
+      SEL: 'idle'
+    },
+    error: {
+      KEY: 'type'
+    }
+  };
+  
+  constructor(props) {
+    super(props);
+      
+    this.state = {
+      machineState: this.initialState,
+      query: undefined,
+      results: undefined,
+      timer: undefined
+    };
+    
+    this.commands = {
+      idle: this.reset,
+      type: this.isTyping,
+      search: this.search,
+      results: noop,
+      error: this.displayError
+    };
+  }
+
+  transition(action) {
+    const { machineState } = this.state;
+    const nextState = this.machine[machineState][action];
+    const command = this.commands[nextState];
+    this.setState({ machineState: nextState }, command);
+  }
+
+  onChange = (query) => {
+    this.setState({ query });
+    this.transition('KEY');
+  }
+
+  onKey = (key) => {
+    if (key === 27) {
+      this.transition('ESC');
+    }
+  }
+
+  onClick = (i) => () => {
+    this.props.onSelect(this.state.results[i]);
+    this.transition('SEL');
+  }
+
+  reset = () => {
+    if (this.state.timer) {
+      clearTimeout(this.state.timer);
+    }
+    
+    this.setState({
+      query: undefined,
+      results: undefined,
+      timer: undefined
+    });
+  }
+
+  isTyping = () => {
+    if (this.state.timer) {
+      clearTimeout(this.state.timer);
+    }
+
+    this.setState({
+      results: undefined,
+      error: undefined,
+      timer: setTimeout(() => { this.transition('TIMEOUT'); },200)
+    });
+  }
+
+  search = () => {
+    if (!(this.state.query && this.state.query.length > 2)) {
+      this.transition('ESC');
+      return;
+    }
+    
+    const query = Object.assign({}, this.props.filters, {
+      q: this.state.query
+    });
+
+    searchAtoms(query)
+      .then(results => {
+        this.setState({ results });
+        this.transition('RESOLVE');
+      })
+      .catch(error => {
+        this.setState({ error });
+        this.transition('REJECT');
+      });
+  }
+
+  displayError = () => {
+
```

Comment:
should something happen here? Not sure how we should communicate this to the user, I did something very basic [here](https://github.com/guardian/atom-workshop/blob/b2001d1484c8cd99768c0e9a7d5bd4d9a75ed656/public/js/components/CommonsDivisions/CommonsDivision.js#L169)

Reply:
oh yeah you're right, totally forgot that 😆 I'm hoping the red band component can be used here. let's see.